cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -2567,17 +2576,28 @@ private CloudServicePlan findPlanForService(CloudService service) {
         }
         Map<String, Object> responseMap = JsonUtil.convertJsonToMap(response);
         List<Map<String, Object>> newResources = (List<Map<String, Object>>) responseMap.get("resources");
-        if (newResources != null && newResources.size() > 0) {
+        if (newResources != null && !newResources.isEmpty()) {
             allResources.addAll(newResources);
         }
         addAllRemainingResources(responseMap, allResources);
         return allResources;
     }
 
     private void addAllRemainingResources(Map<String, Object> responseMap, List<Map<String, Object>> allResources) {
-        String nextUrl = (String) responseMap.get("next_url");
-        while (nextUrl != null && nextUrl.length() > 0) {
-            nextUrl = addPageOfResources(nextUrl, allResources);
+        String nextUrlKey = "next_url";
+        String nextUrl = (String) responseMap.get(nextUrlKey);
```

Comment:
Extract the code that retrieves the next URL in a separate method.

Reply:
Done